### DISCUSSION TITLE
Tag edit panel improvement

### DISCUSSION THREAD

**Discussion by quickfold** [freeplane/freeplane#1992] [Ideas]
Currently, the tags edit panel allows you to select one or more tags and drag them to a node to tag the node. I would also like the ability to tag nodes by selecting one or more nodes and dragging them to a tag. For me, when tagging, there are usually a lot more nodes than relevant tags. So letting the user tag many nodes at once could save a lot of time.

There could also be an option, when dragging nodes to tags, to tag just the selected nodes or to also tag their children/grandchildren/etc.

Lastly, I use tags much more than icons, so I would like the option to have the left tag panel usable for tags in addition to or instead of icons. If that is rejected, having a Tags tab in the tool panel would be better than the current floating window.

**Comment by Illegit** [comment]
+1
Hi, after intensively using tags for many weeks: I support the proposals of quickfold

About the following one:
"having a Tags tab in the tool panel would be better than the current floating window."
I think this is really necessary, I prefer to have it in the tool panel where I use the EdoTools like the Tabbed Menu and the Menu-o-Matic (see the image)
![image](https://github.com/user-attachments/assets/dfbda67e-c510-4be8-bdaf-9f7e069bc2c1)

Thanks

**Reply by dpolivaev** [reply]
This issue is a separate issue and I prefer having one issue per discussion.
Because Tag Category Manager and Tag Editor have "OK" and "Cancel" buttons, is seems natural to have them in dialogs.

**Comment by dpolivaev** [comment]
I am changing the logic so that dragging tags to any selected nodes adds them to all selected nodes

**Reply by dpolivaev** [reply]
The same logic applies to copy and paste

**Reply by dpolivaev** [reply]
@quickfold Whenever it happens to you that you have 3 ideas please start 3 discussions or submit 3 issues because otherwise I can neither understand nor track them properly !!!!!

**Reply by dpolivaev** [reply]
Dragged/pasted tags are added to all selected nodes: Preview 1.12.5_01
